execute/observation.go

```diff
@@ -173,6 +182,8 @@ func (p *Plugin) getCommitReportsObservation(
 		return exectypes.Observation{}, err
 	}
 
+	// TODO: message from fullyExecutedCommits which are in the inflight messages cache could be cleared here.
```

Comment:
Still a relevant TODO?

Reply:
Still relevant. We need some extra message data from the `ExecutionStateChangedEvent`.

Issue for this optimization/TODO: [CCIP-5167](https://smartcontract-it.atlassian.net/browse/CCIP-5167)